lightly/models/utils.py

```diff
@@ -28,23 +28,21 @@
 
 
 def pool_masked(
-    source: Tensor, mask: Tensor, reduce: str = "mean", num_cls: Optional[int] = None
+    source: Tensor, mask: Tensor, num_cls: int, reduce: str = "mean"
 ) -> Tensor:
-    """Reduce image feature maps (B, C, H, W) or (C, H, W) according to an integer
-    index given by `mask` (B, H, W) or (H, W).
+    """Reduce image feature maps :math:`(B, C, H, W)` or :math:`(C, H, W)` according to
+    an integer index given by `mask` :math:`(B, H, W)` or :math:`(H, W)`.
 
     Args:
-        source: Float tensor of shape (B, C, H, W) or (C, H, W) to be reduced.
-        mask: Integer tensor of shape (B, H, W) or (H, W) containing the integer indices.
-        reduce: The reduction operation to be applied, one of 'prod', 'mean', 'amax' or
-            'amin'. Defaults to 'mean'.
-        num_cls: The number of classes in the possible masks. If None, the number of classes
-            is inferred from the unique elements in `mask`. This is useful when not all
-            classes are present in the mask.
+        source: Float tensor of shape :math:`(B, C, H, W)` or :math:`(C, H, W)` to be
+            reduced.
+        mask: Integer tensor of shape :math:`(B, H, W)` or :math:`(H, W)` containing the
+            integer indices.
+        num_cls: The number of classes in the possible masks.
 
     Returns:
-        A tensor of shape (B, C, N) or (C, N) where N is the number of unique elements
-        in `mask` or `num_cls` if specified.
+        A tensor of shape :math:`(B, C, N)` or :math:`(C, N)` where :math:`N`
+        corresponds to `num_cls`.
```

Comment:
Minor: Might make it a bit easier to understand:

```suggestion
        A tensor of shape :math:`(B, C, num_cls)` or :math:`(C, num_cls)`.
```